core/lib/src/config/config.rs

```diff
@@ -85,13 +85,28 @@ pub struct Config {
     /// client. Used internally and by [`Request::client_ip()`] and
     /// [`Request::real_ip()`].
     ///
-    /// To disable using any header for this purpose, set this value to `false`.
-    /// Deserialization semantics are identical to those of [`Ident`] except
-    /// that the value must syntactically be a valid HTTP header name.
+    /// To disable using any header for this purpose, set this value to `false`
+    /// or `None`. Deserialization semantics are identical to those of [`Ident`]
+    /// except that the value must syntactically be a valid HTTP header name.
     ///
     /// **(default: `"X-Real-IP"`)**
-    #[serde(deserialize_with = "crate::config::ip_header::deserialize")]
+    #[serde(deserialize_with = "crate::config::http_header::deserialize")]
     pub ip_header: Option<Uncased<'static>>,
+    /// The name of a header, whose value is typically set by an intermediary
+    /// server or proxy, which contains the protocol (HTTP or HTTPS) used by the
+    /// connecting client. This should probably be [`X-Forwarded-Proto`], as
+    /// that is the de facto standard. Used by [`Request::forwarded_proto()`]
```

Comment:
The wording "this should probably be" begets the question: why isn't this the default, then?

I think we need to be much, much clearer here about what this value control, when it should be set, and if set, what values might be typical.